jeeves/jobs.py

```diff
@@ -55,13 +55,19 @@ def get_jenkins_job_info(server, job_name, filter_param_name=None, filter_param_
 		if filter_param_name is not None and filter_param_value is not None:
 			api_param_value = [param['value'] for param in build_parameters if filter_param_name == param.get('name', '')][0]
 			while api_param_value != filter_param_value:
-				lcb_num = lcb_num - 1
+				if 'previousBuild' in build_info and build_info['previousBuild'] is not None:
+					lcb_num = build_info['previousBuild']['number']
+				else:
+					# there is no previous build available,
+					# set NO_KNOWN_BUILDS found
+					raise Exception("No filter match")
```

Comment:
I'm wondering if this is worth having a custom exception? https://www.programiz.com/python-programming/user-defined-exception

Might be more trouble than it's worth

Reply:
Well, I thought that's it more trouble than it's worth.

Reply:
Fair enough, we'll keep it in the back pocket :)